### PR TITLE
MySQL Java time instances

### PR DESCRIPTION
**Pull request by guymers**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -266,7 +280,7 @@ lazy val postgres = project
       import doobie._, doobie.implicits._
       import doobie.postgres._, doobie.postgres.implicits._
       implicit val cs = IO.contextShift(scala.concurrent.ExecutionContext.global)
-      val xa = Transactor.fromDriverManager[IO]("org.postgresql.Driver", "jdbc:postgresql:world", "postgres", "")
+      val xa = Transactor.fromDriverManager[IO]("org.postgresql.Driver", "jdbc:postgresql:world", "postgres", "password")
```
Comment (jatcwang) [inline build.sbt:283]: Not opposed to this change but can you split this out to a separate PR for me? :slightly_smiling_face:

Comment (guymers) [reply build.sbt:283]: https://github.com/tpolecat/doobie/pull/1792

File: modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      List(JT.VarChar, JT.Date, JT.Time),
+      classOf[java.time.OffsetDateTime]
+    )
+
+  implicit val JavaTimeInstantMeta: Meta[java.time.Instant] =
+    JavaTimeOffsetDateTimeMeta.timap(_.toInstant)(OffsetDateTime.ofInstant(_, ZoneOffset.UTC))
+
+  implicit val JavaTimeLocalDateTimeMeta: Meta[java.time.LocalDateTime] =
+    Basic.oneObject(
+      JT.Timestamp,
```
Comment (sideeffffect) [inline modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala:28]: MySQL/MariaDB has only `TIMESTAMP` which behaves as if `WITH TIME ZONE`, right? It is the equivalent of Java's `Instant`.

So that would mean that `LocalDateTime` is not suitable for `TIMESTAMP`, right?

File: modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+trait JavaTimeInstances extends MetaConstructors {
+
+  implicit val JavaTimeOffsetDateTimeMeta: Meta[java.time.OffsetDateTime] =
+    Basic.oneObject(
+      JT.Timestamp,
+      List(JT.VarChar, JT.Date, JT.Time),
+      classOf[java.time.OffsetDateTime]
+    )
+
+  implicit val JavaTimeInstantMeta: Meta[java.time.Instant] =
```
Comment (sideeffffect) [inline modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala:23]: Shouldn't this be the "primary" Meta instance and the one for `OffsetDateTime` be derived from it?

File: modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala (hunk 2 above)
Comment (guymers) [reply modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala:39]: Yeah those are pretty wrong, should be fixed now.

File: modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala (hunk 3 above)
Comment (guymers) [reply modules/mysql/src/main/scala/doobie/mysql/JavaTimeInstances.scala]: `Instant` isn't mentioned in the JDBC spec so drivers don't have to provide a way to convert a column to/from it. The MySQL driver errors if you ask it for an `Instant` so it is implemented in terms of `OffsetDateTime`.

**Review by sideeffffect — APPROVED** [review]
:heart:

**Comment by sideeffffect** [comment]
@guymers why did you close this? Is it superseded by https://github.com/tpolecat/doobie/pull/1789 ?

**Comment by guymers** [comment]
Its been 5 months and I don't use mysql.

**Comment by jatcwang** [comment]
I'll get this through the finish line. Thanks

**Comment by sideeffffect** [comment]
@guymers should this be merged before or after https://github.com/tpolecat/doobie/pull/1789 ?

**Comment by guymers** [comment]
Doesn't really matter, but may as well do this one.

**Comment by jatcwang** [comment]
Superceded by #2024 after #2023 allowing more precise typechecking based on DB vendor's type name.
@guymers I've added you as a coauthor in the commit. Thanks!
